src/Core/Domain/Product/QueryResult/SpecificPriceForEditing.php

```diff
@@ -0,0 +1,297 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Domain\Product\QueryResult;
+
+use DateTime;
+use PrestaShop\Decimal\DecimalNumber;
+
+class SpecificPriceForEditing
+{
+    /**
+     * @var int
+     */
+    private $specificPriceId;
+
+    /**
+     * @var string
+     */
+    private $reductionType;
+
+    /**
+     * @var DecimalNumber
+     */
+    private $reductionAmount;
+
+    /**
+     * @var bool
+     */
+    private $includeTax;
+
+    /**
+     * @var DecimalNumber
+     */
+    private $price;
+
+    /**
+     * @var int
+     */
+    private $fromQuantity;
+
+    /**
+     * @var int|null
+     */
+    private $shopGroupId;
+
+    /**
+     * @var int|null
+     */
+    private $shopId;
+
+    /**
+     * @var int|null
+     */
+    private $cartId;
+
+    /**
+     * @var int|null
+     */
+    private $currencyId;
+
+    /**
+     * @var int|null
+     */
+    private $catalogPriceRuleId;
+
+    /**
+     * @var int|null
+     */
+    private $countryId;
+
+    /**
+     * @var int|null
+     */
+    private $groupId;
+
+    /**
+     * @var int|null
+     */
+    private $customerId;
+
+    /**
+     * @var DateTime|null
+     */
+    private $dateTimeFrom;
+
+    /**
+     * @var DateTime|null
+     */
+    private $dateTimeTo;
+
+    /**
+     * @param int $specificPriceId
+     * @param string $reductionType
+     * @param DecimalNumber $reductionAmount
+     * @param bool $includeTax
```

Comment:
So `reduction_tax` in database is actually a boolean which means whether or not tax is included 😮 ?

Reply:
Yes more or less, it is used when the reduction is of type amount (since percent doesn't care about taxes) And it indicates if reductionAmount is expressed with or without tax